jlcparts/datatables.py

```diff
@@ -270,56 +273,88 @@ def clearDir(directory):
         elif os.path.isdir(file_path):
             shutil.rmtree(file_path)
 
+
+@dataclasses.dataclass
+class MapCategoryParams:
+    libraryPath: str
+    outdir: str
+    ignoreoldstock: int
+
+    catName: str
+    subcatName: str
+
+
+def _map_category(val: MapCategoryParams):
+    # Sometimes, JLC PCB doesn't fill in the category names. Ignore such
+    # components.
+    if val.catName.strip() == "":
+        return None
+    if val.subcatName.strip() == "":
+        return None
+
+    filebase = val.catName + val.subcatName
+    filebase = filebase.replace("&", "and").replace("/", "aka")
+    filebase = re.sub('[^A-Za-z0-9]', '_', filebase)
+
+    lib = PartLibraryDb(val.libraryPath)
```

Comment:
I've also tested loading the library only once per subprocess, but the difference in performance is in the noise.

- 1x per process: 120.40s user 7.18s system 626% cpu 20.368 total
- 1x per subcategory: 122.83s user 7.66s system 638% cpu 20.422 total